react/src/components/Weather/CurrentWeather/CurrentWeather.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { format } from "date-fns";
+import { pl } from "date-fns/locale";
+import { DisplayData } from "../DataDisplay";
+import {
+  getSunriseTime,
+  getDate,
+  getAqi,
+  getClouds,
+  getFeelsLike,
+  getHumidity,
+  getPressure,
+  getSunsetTime,
+  getTemp,
+  getWindSpeed,
+} from "../Getters";
+import CurrentStyle from "./CurrentW.module.css";
+import sun from "../../../icons/desktop/weathersun.png";
+import rain from "../../../icons/desktop/weatherrain.png";
+import cloud from "../../../icons/desktop/weathercloud.png";
+
+const ICONS = {
+  Clear: sun,
+  Rain: rain,
+  Clouds: cloud,
+};
+const dateFormatPurposueDescription = {
+  date: "cccc, dd/MM/yy, h:mm",
+};
+
+const CurrentWeather = ({ data }) => {
+  if (!data) {
+    return <>N/A</>;
+  }
+
+  return (
+    <div className={CurrentStyle.app}>
+      <div className={CurrentStyle.weather_side}>
+        <div className={CurrentStyle.weather_gradient}>
+          <div className={CurrentStyle.location_box}>
+            <i className={CurrentStyle.location_icon}></i>
+            <span className={CurrentStyle.location}>
+              <DisplayData data={data.name} />
+            </span>
+            <span className={CurrentStyle.date}>
+              {format(new Date(), dateFormatPurposueDescription.date, {
+                locale: pl,
+              })}
+            </span>
+          </div>
+        </div>
+        <div className={CurrentStyle.weather_box}>
+          <div className={CurrentStyle.temp}>
+            <div className={CurrentStyle.weather_description}>
+              {data.weather[0].main}{" "}
+            </div>
+            <div className={CurrentStyle.weather_temp}>
+              <DisplayData data={getTemp(data)} />
+              °C
+            </div>
+            <img
+              className={CurrentStyle.weather_icon}
+              src={ICONS[data.weather[0].main]}
```

Comment:
data.weather[0].main to jest użyte więcej razy, może warto to do zmiennej zapisać :)